### PR TITLE
Fix for Webpa Force sync is set for NULL supplementary URL

### PR DESCRIPTION
**Pull request by guruchandru**

### CODE REVIEW

File: src/webcfg_rbus.c [hunk 1]
```diff
@@ -383,14 +389,18 @@ rbusError_t webcfgFrSetHandler(rbusHandle_t handle, rbusProperty_t prop, rbusSet
...
-				WebcfgError("Telemetry url is null so, force sync SET failed\n");
-				return RBUS_ERROR_BUS_ERROR;
+				char telemetryUrl[256] = {0};
+				Get_Supplementary_URL("Telemetry", telemetryUrl);
+				if(strncmp(telemetryUrl,"NULL",strlen("NULL")) == 0)
+				{
+					WebcfgError("Telemetry url is null so, force sync SET failed\n");
+					return RBUS_ERROR_BUS_ERROR;
+				}
+				telemetryUrlCheck = 1;
```
Comment (shilpa24balaji) [inline src/webcfg_rbus.c:403]: telemetryUrlCheck is not needed as when strlen matches it is string case and not json case. parsejson will pass only for json case

File: src/webcfg_rbus.c [hunk 2]
```diff
@@ -1343,6 +1353,24 @@ int parseForceSyncJson(char *jsonpayload, char **forceSyncVal, char **forceSynct
...
+				{
+					if(0 == strncmp(force_sync_str,"telemetry",strlen("telemetry")))
+					{
+						char telemetryUrl[256] = {0};
+						Get_Supplementary_URL("Telemetry", telemetryUrl);
+						if(strncmp(telemetryUrl,"NULL",strlen("NULL")) == 0)
+						{
+							WebcfgError("Telemetry url is null so, force sync SET failed\n");
+							cJSON_Delete(json);
+							return 1;
```
Comment (shilpa24balaji) [inline src/webcfg_rbus.c:1366]: better to return -1 to indicate error

File: src/webcfg_rbus.c [hunk 3]
```diff
@@ -1382,7 +1411,11 @@ int set_rbus_ForceSync(char* pString, int *pStatus)
 	if(strlen(pString)>0)
 	{
 		WebcfgInfo("Received poke request, proceed to parseForceSyncJson\n");
-		parseForceSyncJson(pString, &value, &transactionId);
+		parseJsonRet = parseForceSyncJson(pString, &value, &transactionId);
+		if(1 == parseJsonRet)
+		{
+			return 0;
```
Comment (shilpa24balaji) [inline src/webcfg_rbus.c:1417]: add comment like 0 is error in function header or here

File: src/webcfg_rbus.c (hunk 1 above)
Comment (guruchandru) [reply src/webcfg_rbus.c]: removed the check in latest push

File: src/webcfg_rbus.c (hunk 2 above)
Comment (guruchandru) [reply src/webcfg_rbus.c]: added the change

File: src/webcfg_rbus.c (hunk 3 above)
Comment (guruchandru) [reply src/webcfg_rbus.c]: added the comment

File: src/webcfg_rbus.c [hunk 4]
```diff
@@ -1343,6 +1346,17 @@ int parseForceSyncJson(char *jsonpayload, char **forceSyncVal, char **forceSynct
 			force_sync_transid = cJSON_GetObjectItem( json, "transaction_id" )->valuestring;
 			if ((force_sync_str != NULL) && strlen(force_sync_str) > 0)
 			{
+				if(0 == strncmp(force_sync_str,"telemetry",strlen("telemetry")))
```
Comment (shilpa24balaji) [inline src/webcfg_rbus.c:1349]: Can you test if json string is telemetry-test

Comment (guruchandru) [reply src/webcfg_rbus.c]: tested and adding strlen check as discussed

**Review by sadhyama — APPROVED** [review]

**Review by shilpa24balaji — APPROVED** [review]

**Pull request merged**